tests/test_url_for.py

```
@@ -0,0 +1,56 @@
+)
+def test_app_url_path_for_page(path_name: str, url: str, path_params: dict[str, Any]) -> None:
+    """
+    Tests that `app.url_path_for()` can construct URLs for pages
+    based on page module name (import path).
+    """
+    assert app.url_path_for(path_name, **path_params) == url
+
+
+@pytest.mark.parametrize(
+    ("path_name", "url"),
+    (("test_app.calculator.page.handle_submit", "/calculator/"),),
```

Ruff (0.14.1)

Wrong values type in `pytest.mark.parametrize` expected `list` of `tuple`

Use `list` of `tuple` for parameter values

(PT007)